### PR TITLE
Fix webhook panic after refresh

### PR DESCRIPTION
**Pull request by mchaynes**
### Description

Fixes https://github.com/pulumi/pulumi-pulumiservice/issues/85

The reason the panic happened was due to a mismatch between the properties produced by `Read` and the properties expected by `Diff`. You can see the difference in output states in more detail below, but the TL;DR is that we were adding an extra object with the key `__inputs` in our output property map upon `Create` and `Update`. `Read` did not stuff values into the `__inputs` object after reading.

In order to simplify things, I lifted every key we previously stored in `__inputs` to the top level, while still handling the old state. So when users upgrade to this new version, on `pulumi up` or `pulumi refresh`, they will see a diff as if we are adding additional output properties, and removing old properties. We don't do this `__inputs` thing in any other providers (See [`pulumi-command`](https://github.com/pulumi/pulumi-command/blob/master/provider/pkg/provider/provider.go#L164) for example)

Additionally, I upgraded examples/go.mod to `go1.19` so that I could upgrade our pinned version of `pulumi/pulumi/sdk` and `pulumi/pulumi/pkg`.

#### Before Refresh:
```json
"inputs": {
    "active": true,
    "displayName": "yaml-webhook",
    "organizationName": "service-provider-test-org",
    "payloadUrl": "https://google.com"
},
"outputs": {
    "__inputs": {
        "active": true,
        "displayName": "yaml-webhook",
        "organizationName": "service-provider-test-org",
        "payloadUrl": "https://google.com"
    },
    "name": "03dbae8b"
},
```
#### After Refresh
```json
"inputs": {
    "__inputs": {
        "active": true,
        "displayName": "yaml-webhook",
        "organizationName": "service-provider-test-org",
        "payloadUrl": "https://google.com"
    },
    "name": "03dbae8b"
},
"outputs": {
    "active": true,
    "displayName": "yaml-webhook",
    "organizationName": "service-provider-test-org",
    "payloadUrl": "https://google.com",
    "secret": ""
},
```

### After This PR
(Still handles old state where `__inputs` was defined)
```json
"inputs": {
    "active": true,
    "displayName": "yaml-webhook",
    "organizationName": "service-provider-test-org",
    "payloadUrl": "https://google.com",
    "secret": ""
},
"outputs": {
    "active": true,
    "displayName": "yaml-webhook",
    "name": "03dbae8b",
    "organizationName": "service-provider-test-org",
    "payloadUrl": "https://google.com"
},
```

### Testing
* Added new test cases to check that refresh works in nodejs and yaml.

##### v0.1.6 --> PR version, with `refresh`
1. Verified installed `v0.1.6` version of the pulumi-service provider.
2. Ran `pulumi up` in `examples/yaml-webhooks`.
3. Ran `pulumi refresh`
4. Upgraded pulumi-service provider to PR version.
5. Ran `pulumi up`
6. Observed `pulumi up` ran successfully and lifted values from `__inputs` to top-level outputs

##### v0.1.6 --> PR version, without `refresh` on v0.1.6
1. Verified installed `v0.1.6` version of the pulumi-service provider.
2. Ran `pulumi up` in `examples/yaml-webhooks`.
4. Upgraded pulumi-service provider to PR version.
5. Ran `pulumi up`
6. Observed `pulumi up` ran successfully and lifted values from `__inputs` to top-level outputs

### CODE REVIEW

File: examples/yaml-webhooks/Pulumi.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  webhook:
+    type: pulumiservice:index:Webhook
+    properties:
+      active: true
+      displayName: yaml-webhook
+      organizationName: service-provider-test-org
+      payloadUrl: "https://google.com"
+
+outputs:
+  # export the value of the access token
```
Comment (mchaynes) [inline examples/yaml-webhooks/Pulumi.yaml:15]:
```suggestion
  # export the name of the webhook
```

File: provider/pkg/internal/pulumiapi/webhooks.go [hunk 2]
```diff
@@ -42,11 +42,11 @@ type updateWebhookRequest struct {
 	OrganizationName string `json:"organizationName"`
 	DisplayName      string `json:"displayName"`
 	PayloadURL       string `json:"payloadUrl"`
-	Secret           string `json:"secret"`
+	Secret           *string `json:"secret"`
```
Comment (blampe) [inline provider/pkg/internal/pulumiapi/webhooks.go:45]: Do you want omitempty here, too?

**Review by blampe — APPROVED** [review]
Ah I figured the `__inputs` was some Pulumi thing, I guess not! Nice cleanup.

File: provider/pkg/internal/pulumiapi/webhooks.go (hunk 2 above)
Comment (mchaynes) [inline provider/pkg/internal/pulumiapi/webhooks.go:45]:
```suggestion
	Secret           *string `json:"secret,omitempty"`
```

Comment (mchaynes) [reply provider/pkg/internal/pulumiapi/webhooks.go]: Good catch, yeah.

**Pull request merged**